runtime/prelude.jakt

```diff
@@ -22,6 +22,7 @@ extern struct Array<T> {
     function iterator(this) -> ArrayIterator<T>
     function first(this) -> T?
     function last(this) -> T?
+    function join(this, anon s: String) throws -> String
```

Comment:
```suggestion
    function join(this, anon separator: String) throws -> String
```
Would be more consistent and expressive

Reply:
Yeah, if preferable could change this. There is kind of a mix currently in descriptive names and one letter names for anon parameters in prelude currently.
In effect no one would actually use the parameter name in this case.

Reply:
still adding more code that violates the code style also isn't great

Reply:
Sure makes sense, I have updated them.